float32 face_depth
float32 face_x
float32 face_y
float32 face_angle
float32 depth_diff